dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/DotFolderTransformerImpl.java

```diff
@@ -0,0 +1,157 @@
+package com.dotmarketing.portlets.contentlet.transform;
+
+import static com.dotmarketing.business.PermissionAPI.PERMISSION_READ;
+
+import com.dotmarketing.beans.IconType;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.util.Logger;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import com.liferay.portal.model.User;
+import com.liferay.util.StringPool;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Folder Transformers Class
+ */
+public class DotFolderTransformerImpl implements DotMapViewTransformer {
+
+    @Override
+    public List<Map<String, Object>> toMaps() {
+        return transform();
+    }
+
+    private enum TargetView {
+        BROWSE_VIEW,
+        GRAPHQL_VIEW
+    }
+    private PermissionAPI permissionAPI;
+
+    private final User user;
+    private final Role[] roles;
+    private final List<Folder> folders;
+    private final TargetView targetView;
+
+    /**
+     * Test friendly "private" constructor
+     * @param permissionAPI
+     * @param user
+     * @param folders
+     */
+    @VisibleForTesting
+    DotFolderTransformerImpl(
+            final PermissionAPI permissionAPI,
+            final User user,
+            final Role[] roles,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The user-supplied array 'roles' is stored directly.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5596402)